File: JavaScript/7js 2/sortByLength.js
/*
|--------------------------------------------------------------------------
| Sort array by string length
|--------------------------------------------------------------------------
|
*/

function sortByLength(array) {
   let res = [];
   let x = array.map((i) => i.length).sort((a, b) => a - b);
   for (let i = 0; i < x.length; i++) {
      for (let j = 0; j < array.length; j++) {
         if (array[j].length == x[i]) res.push(array[j]);
      }
   }
   return res;
}


function sortByLength(array) {
   return array.sort((a, b) => a.length - b.length);
}


function sortByLength(array) {
   let shortestWord = array.sort(function (a, z) {
      return a.length - z.length;
   });
   return shortestWord;
}


console.log(sortByLength(["Beg", "Life", "I", "To"]), ["I", "To", "Beg", "Life"]);
console.log(sortByLength(["", "Moderately", "Brains", "Pizza"]), ["","Pizza","Brains","Moderately",]);
console.log(sortByLength(["Longer", "Longest", "Short"]), ["Short", "Longer", "Longest"]);

File: Python/7py 3/sort_by_length.py
"""
|--------------------------------------------------------------------------
| Sort array by string length
|--------------------------------------------------------------------------
|
"""

def sort_by_length(arr):
   res = []
   x = sorted(list(map(lambda i: len(i), arr)))
   for i in x:
      for j in arr:
         if len(j) == i: res.append(j)
   return res


def sort_by_length(arr):
   return sorted(arr, key=len)


def sort_by_length(arr):
   words = {}
   for word in arr:
      words[int(len(word))]=word
   return [words[key] for key in sorted(words.keys())]


print(sort_by_length(["Beg", "Life", "I", "To"]), ["I", "To", "Beg", "Life"])
print(sort_by_length(["", "Moderately", "Brains", "Pizza"]), ["","Pizza","Brains","Moderately",])
print(sort_by_length(["Longer", "Longest", "Short"]), ["Short", "Longer", "Longest"])